src/test/java/com/google/sps/ImageSelectionTest.java

```diff
@@ -0,0 +1,75 @@
+package com.google.sps.image;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import java.io.IOException;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+@RunWith(JUnit4.class)
+public final class ImageSelectionTest {
+
+  private final Set<String> keywords = new LinkedHashSet<>();
+
+  // This is how every Bing image link starts
+  private static final String EXPECTED_LINK = "https://tse";
+
+  private void checkScrapingOutputLink() {
+    ImageSelection imageSelection = new ImageSelection(keywords);
+
+    try {
+      String actualLink = imageSelection.getBestImage();
+      Assert.assertEquals(EXPECTED_LINK, actualLink.substring(0, 11));
+    } catch(IOException ex) {
+      // Test fail because the function threw an exception
+      Assert.fail(ex.getMessage());
+    }
+  }
+
+  @Test
+  public void singleKeyword() {
+    keywords.add("london");
+
+    checkScrapingOutputLink();
```

Comment:
Instead of passing information through instance variable (keywords), why not passing the keywords as a function down the stack? This makes it much easier to understand that checkScrapingOutputLink method inspects the content of keywords.